Mathlib/Data/Nat/Prime/MersenneFermat.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2024 Ralf Stephan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ralf Stephan, Thomas Browning
+-/
+import Mathlib.Data.Nat.Prime.Basic
+import Mathlib.Data.Nat.Factorization.Basic
+import Mathlib.Data.ZMod.Basic
+import Mathlib.FieldTheory.Finite.Basic
+import Mathlib.NumberTheory.LegendreSymbol.QuadraticReciprocity
+import Mathlib.NumberTheory.LucasPrimality
+
+/-!
+# Lemmas about Fermat numbers
+-/
+
+open ZMod Nat
+
+/-- Prime `a ^ n + 1` implies `n` is a power of two (**Fermat primes**). -/
+theorem pow_of_pow_add_prime {a n : ℕ} (ha : 1 < a) (hn : n ≠ 0) (hP : (a ^ n + 1).Prime) :
```

Comment:
Probably these should all go into the `Nat` namespace; sorry for not catching that earlier.

Reply:
Put them in a Nat namespace and moved them to the new `Fermat.lean` after talking with @rwst